tests/integration/test_ols.py

```diff
@@ -158,6 +159,52 @@ def test_post_question_on_invalid_question() -> None:
         assert response.json() == expected_json
 
 
+def test_post_question_on_generic_response_type_summarize_error() -> None:
+    """Check the REST API /v1/query with POST HTTP method when generic response type is returned."""
+    # let's pretend the question is valid and generic one
+    answer = (constants.SUBJECT_VALID, constants.CATEGORY_GENERIC)
+    with patch(
+        "ols.app.endpoints.ols.QuestionValidator.validate_question", return_value=answer
+    ):
+        with patch(
+            "ols.app.endpoints.ols.DocsSummarizer.summarize",
+            side_effect=Exception("summarizer error"),
+        ):
+            conversation_id = Utils.get_suid()
+            response = client.post(
+                "/v1/query",
+                json={"conversation_id": conversation_id, "query": "test query"},
+            )
+            assert response.status_code == requests.codes.internal_server_error
+            expected_json = {"detail": "Error while obtaining answer for user question"}
+            assert response.json() == expected_json
+
+
+def test_post_question_on_generic_response_llvm_configuration_error() -> None:
+    """Check the REST API /v1/query with POST HTTP method when generic response type is returned."""
+    # let's pretend the question is valid and generic one
+    answer = (constants.SUBJECT_VALID, constants.CATEGORY_GENERIC)
+    with patch(
+        "ols.app.endpoints.ols.QuestionValidator.validate_question", return_value=answer
+    ):
+        with patch(
+            "ols.app.endpoints.ols.DocsSummarizer.summarize",
+            side_effect=LLMConfigurationError("LLM configurtion error"),
+        ):
+            conversation_id = Utils.get_suid()
+            response = client.post(
+                "/v1/query",
+                json={"conversation_id": conversation_id, "query": "test query"},
+            )
+            assert response.status_code == requests.codes.unprocessable
+            expected_json = {
+                "detail": {
+                    "response": "Unable to process this request because 'LLM configurtion error'"
```

Comment:
```suggestion
                    "response": "Unable to process this request because 'LLM configuration error'"
```

and obviously fix it in the actual error also if it's incorrect there

Reply:
fixed